### PR TITLE
Define and use unsafe_convert instead of manual unwrapping of pointers.

### PR DESCRIPTION
**Pull request by andreasnoack**
Remove `get` function and add check for null pointer to `unsafe_convert`

### CODE REVIEW

File: base/sparse/higherorderfns.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Base.Broadcast: _containertype, promote_containertype,
 
 using Base: front, tail, to_shape
 using ..SparseArrays: SparseVector, SparseMatrixCSC, AbstractSparseVector,
-                      AbstractSparseMatrix, AbstractSparseArray, indtype
+                      AbstractSparseMatrix, AbstractSparseArray, indtype, nnz, nzrange
```
Comment (tkelman) [inline base/sparse/higherorderfns.jl]: seems unrelated?

**Comment by andreasnoack** [comment]
Fixing the test failure conflicts with #22626 so I'll rebase when that has been merged

File: base/sparse/higherorderfns.jl (hunk 1 above)
Comment (andreasnoack) [inline base/sparse/higherorderfns.jl]: These are missing imports so they fail when reloading the `Sparse` module. They are exported from `Base` and therefore automatically imported the first time but not on reloads.

Comment (tkelman) [inline base/sparse/higherorderfns.jl]: that seems like a bug in reloading behavior

Comment (andreasnoack) [inline base/sparse/higherorderfns.jl]: @vtjnash Is this a bug or shouldn't we expect the special behavior of Base, Main, and Core to carry through to `eval(:(include))`

Comment (tkelman) [inline base/sparse/higherorderfns.jl]: these really shouldn't be necessary
